### PR TITLE
Activities tab

### LINKED ISSUE #145: Weekly Activity
How about a weekly activity mouseover or dropdown?  Quick example below thrown together in paint.  Something that would be convenient, but not always shown. 

quick example

### PR DESCRIPTION
**Pull request by kyleshay**
Figured we should get this into DIM as seeing these things within the same app might be useful for D2.

I’m intentionally hiding some activities (by using a whitelist, because is it really that important to know what the weekly crucible type is? you can just see it in game. weekly strike/nightfall is nice because then you can setup your loadout based on the burn.) I can add more in, but for the first phase this is what is shown.

screen shot 2017-07-05 at 10 55 23 pm

todo: 

- [x] nightfall still needs a checkbox
- [x] weekly heroic doesn't need to show each of the characters, just the modifiers.
~~- [ ] weekly heroic should show number of weekly bonuses remaining~~
- [x] i18n

This fixes #145

### CODE REVIEW

**Comment by kyleshay** [comment]
screen shot 2017-07-05 at 10 55 23 pm

**Comment by kyleshay** [comment]
nightfall still needs a checkbox, and the weekly heroic doesn't need to show each of the characters, just the modifiers.

**Comment by bhollis** [comment]
Whoa, fantastic!

**Comment by kyleshay** [comment]
updated UI -- open for comments and feedback 

screen shot 2017-07-06 at 1 23 30 am

**Comment by kyleshay** [comment]
Did as much i18n as I could. Unfortunately raw response with the strike modifiers don't have a hash on them, so I can't look them up in the definitions table!

screen shot 2017-07-06 at 2 42 22 am

File: src/scripts/activities/activities.component.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        .filter((a) => {
+          return whitelist.includes(a.identifier);
+        })
+        .sortBy((a) => {
+          const ix = whitelist.indexOf(a.identifier);
+          return (ix === -1) ? 999 : ix;
+          //          return -a.display.categoryHash;
+        })
+        .map((a) => processActivities(defs, stores, a))
+        .value();
```
Comment (bhollis) [inline src/scripts/activities/activities.component.js]: Just a heads up, I'm going to be removing usages of `chain` in hopes of maintaining future compatibility with Lodash (which has chain, but is somewhat inefficient bundle-size-wise with it).

File: src/scripts/activities/activities.component.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+    const activity = {
+      hash: rawActivity.display.activityHash,
+      name: categoryDef.title,
+      icon: rawActivity.display.icon,
+      image: rawActivity.display.image,
+    };
+
+    if (rawActivity.extended) {
+      activity.skulls = rawActivity.extended.skullCategories.map((s) => {
```
Comment (bhollis) [inline src/scripts/activities/activities.component.js]: This is a good place to use `flatMap`.

File: src/scripts/activities/activities.component.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      }).reduce((a, b) => {
+        return a.concat(b);
+      });
+      //      .map((s) => {
+      //        return skullDef.get(s.hash);
+      //      });
+    }
+
+    if (rawActivity.activityTiers[0].skullCategories && rawActivity.activityTiers[0].skullCategories.length) {
+      activity.skulls = rawActivity.activityTiers[0].skullCategories[0].skulls;
```
Comment (bhollis) [inline src/scripts/activities/activities.component.js]: Pull the skullCategories into a local.

File: src/scripts/activities/activities.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<back-link></back-link>
+
+<div class="activities">
+  <div class="activity" ng-repeat="activity in $ctrl.activities track by $index">
+
+    <div class="activity-header">
+      <div class="activity-title" style="background-image: url({{ ::activity.image | bungieIcon }})" ng-class="{ 'activity-featured': activity.featured }">
```
Comment (bhollis) [inline src/scripts/activities/activities.html]: We have a `bungieBackground` filter for this.

File: src/scripts/activities/activities.scss [hunk 5]
```diff
@@ -0,0 +1,91 @@
+back-link {
+  width: 1160px;
+  margin: 0 auto;
+  display: block;
+}
+
+.activities {
+  width: 1200px;
```
Comment (bhollis) [inline src/scripts/activities/activities.scss]: We should centralize a style for "wide" pages so when we have to do mobile we can change them all at once. In the meantime, maybe this should be `max-width`?

Comment (kyleshay) [inline src/scripts/activities/activities.scss]: swapped to max-width for now

**Comment by kyleshay** [comment]
Thanks for the feedback! I'll do a little more testing and merge to beta for more feedback soon.

**Comment by bhollis** [comment]
Yeah it'd be great to merge it in.

**Pull request merged**